cardano-api-test/test/Test/Cardano/Api/TxBody.hs

```diff
@@ -0,0 +1,263 @@
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module Test.Cardano.Api.TxBody (tests) where
+
+import           Cardano.Prelude
+
+import           Hedgehog (forAll, property, tripping)
+import           Test.Tasty (TestTree, testGroup)
+import           Test.Tasty.Hedgehog (testProperty)
+import           Test.Tasty.TH (testGroupGenerator)
+
+import           Cardano.Api
+
+import           Gen.Cardano.Api.Typed (genTxBody, genTxBodyContent)
+
+
+test_roundtrip_TxBody_make_get :: [TestTree]
+test_roundtrip_TxBody_make_get =
+  [ testProperty (show era) $
+    property $ do
+      content <- forAll $ genTxBodyContent era
+      tripping
+        (normalizeOriginal $ review content)
+        (\_ -> makeTransactionBody content)
+        (<&> \(TxBody content') -> normalizeRoundtrip content')
+  | AnyCardanoEra era <- allCardanoEras
+  ]
+
+
+-- * normalize = strip unnecessary details
+--
+-- After roundtrip, @Just mempty@ may become @Nothing@ or vice versa.
+-- Input data also may be generated as either @Just 0@ or @Nothing@.
+
+normalizeOriginal :: TxBodyContent ViewTx era -> TxBodyContent ViewTx era
+normalizeOriginal content =
+  content
+    { txAuxScripts     = normalizeAuxScripts     $ txAuxScripts     content
+    , txCertificates   = normalizeCertificates   $ txCertificates   content
+    , txIns            = sortOn fst              $ txIns            content
+    , txInsCollateral  = normalizeInsCollateral  $ txInsCollateral  content
+    , txMetadata       = normalizeMetadata       $ txMetadata       content
+    , txMintValue      = normalizeMintValue      $ txMintValue      content
+    , txUpdateProposal = normalizeUpdateProposal $ txUpdateProposal content
+    , txWithdrawals    = normalizeWithdrawals    $ txWithdrawals    content
+    }
+
+normalizeRoundtrip :: TxBodyContent ViewTx era -> TxBodyContent ViewTx era
+normalizeRoundtrip content@TxBodyContent{txAuxScripts, txIns, txInsCollateral} =
+  content
+    { txAuxScripts    = normalizeAuxScripts    txAuxScripts
+    , txIns           = sortOn fst             txIns
+    , txInsCollateral = normalizeInsCollateral txInsCollateral
+    }
+
+normalizeInsCollateral :: TxInsCollateral era -> TxInsCollateral era
+normalizeInsCollateral = \case
+  TxInsCollateralNone -> TxInsCollateralNone
+  -- for original
+  TxInsCollateral _ ins | null ins -> TxInsCollateralNone
+  -- for roundtrip
+  TxInsCollateral support ins -> TxInsCollateral support $ sort ins
+
+normalizeMetadata :: TxMetadataInEra era -> TxMetadataInEra era
+normalizeMetadata = \case
+  TxMetadataInEra _ (TxMetadata m) | null m -> TxMetadataNone
+  other                                     -> other
+
+normalizeAuxScripts :: TxAuxScripts era -> TxAuxScripts era
+normalizeAuxScripts = \case
+  TxAuxScripts _       []      -> TxAuxScriptsNone
+  TxAuxScripts support scripts ->
+    -- sorting uses script versions, hence sort after upgrade
+    TxAuxScripts support $ sort $ map upgradeScriptInEra scripts
```

Comment:
So this is where you are using the new `Ord` instance on the `ScriptInEra` type. If this just needs to be sorted by the script language/version then you don't need that `Ord` instance (which as you found, is quite complicated and needs lots of other `Ord` instances). I think all you would need here is to use `sortBy` and to use a comparison function that uses the `AnyScriptLanguage` type which already has an `Ord` instance.